Replace split-squeeze pattern

devashishshankar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #101767
* #101766
* __->__ #101765
* #101764

Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

devashishshankar committed:
Replace split-squeeze pattern

Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

[ghstack-poisoned]

devashishshankar committed:
Update on "Replace split-squeeze pattern"


Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

[ghstack-poisoned]

devashishshankar committed:
Update on "Replace split-squeeze pattern"


Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

[ghstack-poisoned]

devashishshankar commented:
@pytorchbot label "topic: not user facing"

devashishshankar committed:
Update on "Replace split-squeeze pattern"


Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

jansel commented on torch/_inductor/fx_passes/split_cat.py:
```diff
@@ -586,6 +593,89 @@ def erase_old_nodes(self, graph, split_node, next_users):
+                        continue
+                    for node in other_node.users:
+                        if node not in searched:
+                            if not matching_nodes or self._match_fns(node):
```
Can you explain what matching_nodes does?

devashishshankar replied:
If you have a better way of doing this please suggest. I was initially thinking of using a different method like `find_sibling_nodes`, but it seemed very similar to `find_anchor_nodes` in implementation.

devashishshankar replied:
btw, I just realized I forgot to add this variable to some other classes. I can add them, but waiting for your response first.

jansel replied:
This isn't really what `find_anchor_nodes()` is meant for, I worry this confuses the semantics.

I'd rather see an extra check, either in the `_match` function or in the handler that is just based on the users of split.

devashishshankar replied:
Sure, I can do that

devashishshankar committed:
Update on "Replace split-squeeze pattern"


Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

devashishshankar committed:
Update on "Replace split-squeeze pattern"


Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

devashishshankar committed:
Update on "Replace split-squeeze pattern"


Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

devashishshankar committed:
Update on "Replace split-squeeze pattern"


Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

devashishshankar committed:
Update on "Replace split-squeeze pattern"


Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

devashishshankar committed:
Update on "Replace split-squeeze pattern"


Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

devashishshankar committed:
Update on "Replace split-squeeze pattern"


Replaces split-squeeze (same dimension) with an unbind. This will be used in combination with later patterns to remove the unbind if it follows a cat/stack

Differential Revision: [D45758181](https://our.internmc.facebook.com/intern/diff/D45758181/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]
